hydroflow_lang/src/graph/mod.rs

```diff
@@ -350,11 +370,13 @@ impl Ord for PortIndexValue {
 pub fn build_hfcode(
     hf_code: HfCode,
     root: &TokenStream,
+    macro_invocation_path: PathBuf,
 ) -> (Option<(HydroflowGraph, TokenStream)>, Vec<Diagnostic>) {
-    let flat_graph_builder = FlatGraphBuilder::from_hfcode(hf_code);
+    let flat_graph_builder = FlatGraphBuilder::from_hfcode(hf_code, macro_invocation_path);
     let (mut flat_graph, uses, mut diagnostics) = flat_graph_builder.build();
-    eliminate_extra_unions_tees(&mut flat_graph);
     if !diagnostics.iter().any(Diagnostic::is_error) {
+        flat_graph.merge_modules();
```

Comment:
Call `merge_modules` be called right when the import is handled?

Reply:
I think it needs to happen after name resolution.